whylog/constraints/__init__.py

```diff
@@ -1,42 +1,51 @@
-from abc import ABCMeta, abstractmethod
+from abc import ABCMeta
 
 import six
 
 
 @six.add_metaclass(ABCMeta)
 class AbstractConstraint(object):
-    @abstractmethod
-    def save(self, rulebase_rule):
-        pass
+    pass
 
 
 class TimeConstraint(AbstractConstraint):
-    def __init__(self, line_earlier, line_later, min_delta=None, max_delta=None):
+    def __init__(self, params):
         pass
 
+    @classmethod
+    def verify(cls, param_dict, group_contents):
+        # TODO remove mock
+        return True
+
 
 class IdenticalIntervals(AbstractConstraint):
-    def __init__(self, intervals):
+    def __init__(self, params):
         pass
 
+    @classmethod
+    def verify(cls, param_dict, group_contents):
+        if not group_contents:
```

Comment:
Because Ewa is editing the same code, I suggested her to raise excepton in such situation.